astropy/table/table.py

```diff
@@ -2104,9 +2105,10 @@ def __setitem__(self, item, value):
                 ):
                     try:
                         self._replace_column_warnings(item, value)
-                        return
                     except Exception:
                         pass
+                    else:
+                        return
```

Comment:
@nstarman, a note for a future PR: I think this can be refactored into

```python
with contexlib.suppress(Exception):
    self._replace_column_warnings(item, value)
    return
```

Reply:
I don't think that is true - the exception passes on to the `self.columns` assignment below.

Indeed, that is actually clearer with the return inside the `try` part - I don't think the rule improves things at all here.